dbms/src/Functions/FunctionsStringSearch.h

```diff
@@ -126,6 +211,9 @@ class FunctionsStringSearch : public IFunction
         const auto * col_haystack_const = typeid_cast<const ColumnConst *>(&*column_haystack);
         const auto * col_needle_const = typeid_cast<const ColumnConst *>(&*column_needle);
 
+        if constexpr (name == std::string_view(NameIlike3Args::name))
+            lowerEnglishWords(block, arguments);
```

Comment:
It maybe not efficient？Why not ignore case during comparing each char?

Reply:
> It maybe not efficient？Why not ignore case during comparing each char?

Good point, but I think it is really hard to implement